src/types.ts

```diff
@@ -412,6 +421,20 @@ export interface VM extends VMXML {
     snapshots: VMSnapshot[] | -1;
 }
 
+/** "Fake" VMs for the UI only **/
+
+export interface UIVM {
```

Comment:
So this is VM which is being installed / created? Maybe we can name that better then UI VM

* VMInstallState
* VMSetupState?

Reply:
Installation has a real VM, and creation and downloading of images for the installation has a fake VM, I think.

Dunno about the name. The existing code uses `isUI` to identify them, and functions like `addUiVm`, `deleteUiVM` to manage them. So I would go with `UIVM`.

Reply:
Right a FakeVM, so UIVM to me doesn't really sound like "Fake".